### PR TITLE
Add support for hexagon-unknown-none-elf as target

### PR DESCRIPTION
**Pull request by androm3da**
Still TODO: document usage details for new target

### CODE REVIEW

**Comment by ehuss** [comment]
r? compiler

File: compiler/rustc_target/src/spec/hexagon_unknown_none_elf.rs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+            linker_flavor: LinkerFlavor::Gnu(Cc::Yes, Lld::No),
+            relocation_model: RelocModel::Pic,
+            position_independent_executables: true,
+            panic_strategy: PanicStrategy::Abort,
+            features: "-small-data,+hvx-length128b".into(),
+            has_thread_local: true,
+            max_atomic_width: Some(32),
+            emit_debug_gdb_scripts: false,
+            c_enum_min_bits: Some(8),
+            ..Default::default()
```
Comment (petrochenkov) [inline compiler/rustc_target/src/spec/hexagon_unknown_none_elf.rs:28]: `linker_flavor` and `relocation_model` are redundant, they already have default values.

**Comment by petrochenkov** [comment]
@androm3da 
Could you do `@rustbot ready` when all the FIXMEs/TODOs are addressed?
The PR is being updated from time to time, but I'm not sure whether I'm expected to review or not yet.

**Comment by androm3da** [comment]
> @androm3da Could you do `@rustbot ready` when all the FIXMEs/TODOs are addressed? The PR is being updated from time to time, but I'm not sure whether I'm expected to review or not yet.

@petrochenkov you bet, sorry for being unclear.  I think this PR is dependent on https://github.com/rust-lang/compiler-builtins/pull/556 in order to be useful.  I'll work on getting that landed and then will address the FIXMEs/TODOs and mark this one `@rustbot ready`.

**Pull request merged**